### ISSUE TITLE
t/thread.t fails on i686 and armv7hl

### ISSUE THREAD

**Issue by jplesnik** [PDLPorters/pdl#311]
Since PDL 2.041, t/thread.t is failing on i686 and armv7hl

```
realloc(): invalid next size
t/thread.t ............. 
All 35 subtests passed 
```
Previous release 2.040 were not possible to build on these architectures.

Info about failing builds can be found on 
https://koji.fedoraproject.org/koji/taskinfo?taskID=66957403

**Comment by mohawk2** [comment]
Using valgrind I've identified what may well have caused this; I changed `threadid` to a `PDL_Indx` but there was one `malloc` which still used `sizeof(int)`. I believe this is fixed by bee5a35dca7124d4502968a645786c61732062e3

**Comment by mohawk2** [comment]
I have released 2.044. @jplesnik Can you confirm whether that then builds on those platforms, and if so close this?

**Comment by jplesnik** [comment]
It is fixed now. Thank you.
